includes/class-wc-subscriptions-email-notifications.php

```diff
@@ -0,0 +1,309 @@
+
+		// Add settings UI.
+		add_filter( 'woocommerce_subscription_settings', [ __CLASS__, 'add_settings' ], 20 );
+
+		add_action( 'update_option_' . WC_Subscriptions_Admin::$option_prefix . self::$offset_setting_string, [ 'WC_Subscriptions_Email_Notifications', 'set_notification_settings_update_time' ] );
+		add_action( 'update_option_' . WC_Subscriptions_Admin::$option_prefix . self::$switch_setting_string, [ 'WC_Subscriptions_Email_Notifications', 'set_notification_settings_update_time' ] );
+		add_action( 'add_option_' . WC_Subscriptions_Admin::$option_prefix . self::$offset_setting_string, [ 'WC_Subscriptions_Email_Notifications', 'set_notification_settings_update_time' ] );
+		add_action( 'add_option_' . WC_Subscriptions_Admin::$option_prefix . self::$switch_setting_string, [ 'WC_Subscriptions_Email_Notifications', 'set_notification_settings_update_time' ] );
+	}
+
+	/**
+	 * Map and forward Edit order screen action to the correct reminder.
+	 *
+	 * @param $order
+	 *
+	 * @return void
+	 */
+	public static function forward_action( $order ) {
+		$trigger_action = '';
+		$current_action = current_action();
+		switch ( $current_action ) {
+			case 'woocommerce_order_action_wcs_customer_notification_free_trial_expiration':
+				$trigger_action = 'woocommerce_scheduled_subscription_customer_notification_trial_expiration';
+				break;
+			case 'woocommerce_order_action_wcs_customer_notification_subscription_expiration':
+				$trigger_action = 'woocommerce_scheduled_subscription_customer_notification_expiration';
+				break;
+			case 'woocommerce_order_action_wcs_customer_notification_renewal':
+				$trigger_action = 'woocommerce_scheduled_subscription_customer_notification_renewal';
+				break;
+		}
+
+		if ( $trigger_action ) {
+			do_action( $trigger_action, $order->get_id() );
+		}
+	}
+
+	/**
+	 * Sets the update time when any of the settings that affect notifications change and triggers update of subscriptions.
+	 *
+	 * When time offset or global on/off switch change values, this method gets triggered and it:
+	 * 1. Updates the wcs_notification_settings_update_time option so that the code knows which subscriptions to update
+	 * 2. Triggers rescheduling/unscheduling of existing notifications.
+	 * 3. Adds a notice with info about the actions that got triggered to the store manager.
+	 *
+	 * @return void
+	 */
+	public static function set_notification_settings_update_time() {
+		update_option( 'wcs_notification_settings_update_time', time() );
+
+		// Shortcut to unschedule all notifications more efficiently instead of processing them subscription by subscription.
+		if ( ! self::notifications_globally_enabled() ) {
+			\WC_Subscriptions_Core_Plugin::instance()->notifications_scheduler->unschedule_all_notifications();
+
+			$message = __( 'Unscheduling all notifications now.', 'woocommerce-subscriptions' );
+		} else {
+			$message = WCS_Notifications_Batch_Processor::enqueue();
+		}
+
+		//TODO: make sure this works first.
```

Comment:
FYI I fixed that [here](https://github.com/Automattic/woocommerce-subscriptions-core/pull/696/files#diff-7c3173cd33df01d8ffe14586bacc988f8dc4df5ba26f34ad6415f69c21f0271dR94)

Reply:
Thanks!